### PR TITLE
Remove duplicate judgment for debug log enable

### PR DESCRIPTION
**Pull request by ChenglongGao327**
Function Debugf has the same judgment on enable debug as DebugEnabled

```
func (s *Scope) Debugf(args ...interface{}) {
	if s.GetOutputLevel() >= DebugLevel {
            ......
	}
}

func (s *Scope) DebugEnabled() bool {
	return s.GetOutputLevel() >= DebugLevel
}
```

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ √] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/security/authz/builder/extauthz.go [hunk 1]
```diff
@@ -100,9 +100,7 @@ func processExtensionProvider(in *plugin.InputParams) map[string]*builtExtAuthz
 		resolved[config.Name] = parsed
 	}
 
-	if authzLog.DebugEnabled() {
-		authzLog.Debugf("Resolved extension providers: %v", spew.Sdump(resolved))
-	}
+	authzLog.Debugf("Resolved extension providers: %v", spew.Sdump(resolved))
```
Comment (hzxuzhonghu) [inline pilot/pkg/security/authz/builder/extauthz.go:103]: IIUC, this change will make `spew.Sdump(resolved)` run first

File: pilot/pkg/xds/ads.go [hunk 2]
```diff
@@ -900,6 +900,8 @@ func (s *DiscoveryServer) removeCon(conID string) {
 
 // Send with timeout if configured.
 func (conn *Connection) send(res *discovery.DiscoveryResponse) error {
+	log.Debugf("discovery response: %s", res)
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/ads.go:903]: unrelated

File: pilot/pkg/xds/delta.go [hunk 3]
```diff
@@ -239,6 +239,8 @@ func (s *DiscoveryServer) receiveDelta(con *Connection) {
 }
 
 func (conn *Connection) sendDelta(res *discovery.DeltaDiscoveryResponse) error {
+	log.Debugf("delta discovery response: %s", res)
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/delta.go:242]: ditto

File: pilot/pkg/xds/xdsgen.go [hunk 4]
```diff
@@ -135,18 +135,16 @@ func (s *DiscoveryServer) pushXds(con *Connection, push *model.PushContext,
 
 	switch {
 	case logdata.Incremental:
-		if log.DebugEnabled() {
-			log.Debugf("%s: %s%s for node:%s resources:%d size:%s%s",
-				v3.GetShortType(w.TypeUrl), ptype, req.PushReason(), con.ConID, len(res), util.ByteCount(configSize), info)
-		}
+		log.Debugf("%s: %s%s for node:%s resources:%d size:%s%s",
```
Comment (hzxuzhonghu) [inline pilot/pkg/xds/xdsgen.go:138]: revert

**Review by howardjohn — COMMENTED** [review]
I believe this is less efficient in many case as it requires us to compute the full arguments. With the conditional we only compute I'd it's required. This is important for some costly logs

File: pilot/pkg/security/authz/builder/extauthz.go (hunk 1 above)
Comment (ChenglongGao327) [reply pilot/pkg/security/authz/builder/extauthz.go]: done

File: pilot/pkg/xds/xdsgen.go (hunk 4 above)
Comment (ChenglongGao327) [reply pilot/pkg/xds/xdsgen.go]: done

File: pilot/pkg/xds/ads.go (hunk 2 above)
Comment (ChenglongGao327) [reply pilot/pkg/xds/ads.go]: Let me delete it

File: pilot/pkg/xds/delta.go (hunk 3 above)
Comment (ChenglongGao327) [reply pilot/pkg/xds/delta.go]: done

File: security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go [hunk 5]
```diff
@@ -240,10 +240,8 @@ func (p *Plugin) constructFederatedTokenRequest(parameters security.StsRequestPa
 		dReq, _ := http.NewRequest("POST", federatedTokenEndpoint, bytes.NewBuffer(dJSONQuery))
 		dReq.Header.Set("Content-Type", contentType)
 
-		if pluginLog.DebugEnabled() {
-			reqDump, _ := httputil.DumpRequest(dReq, true)
-			pluginLog.Debugf("Prepared federated token request: \n%s", string(reqDump))
-		}
+		reqDump, _ := httputil.DumpRequest(dReq, true)
+		pluginLog.Debugf("Prepared federated token request: \n%s", string(reqDump))
```
Comment (hzxuzhonghu) [inline security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go:244]: This is not efficient

Comment (ChenglongGao327) [reply security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go:244]: Repeat, it has been judged at the entry of the code block

Comment (ChenglongGao327) [reply security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go:244]: > This is not efficient



> This is not efficient

Repeat, it has been judged at the entry of the code block

Comment (hzxuzhonghu) [reply security/pkg/stsservice/tokenmanager/google/tokenexchangeplugin.go:244]: noticed

**Review by hzxuzhonghu — COMMENTED** [review]
/ok-to-test

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]
